src/ember-app.js

```diff
@@ -66,7 +68,23 @@ class EmberApp {
 
     // Ensure that the dist files can be evaluated and the `Ember.Application`
     // instance created.
-    this.buildApp();
+    this.buildSandboxQueue(options.maxSandboxQueue);
+  }
+
+  /**
+   * @private
+   *
+   * Function to build queue of sandboxes which is later leveraged if application is using `buildSandboxPerRequest`
+   * flag. This is an optimization to help with performance.
+   *
+   * @param {Number} maxSandboxQueueSize - maximum size of queue (this is should be a derivative of your QPS)
+   */
+  buildSandboxQueue(maxSandboxQueueSize = 1) {
+    this._sandboxApplicationInstanceQueue = new Queue(maxSandboxQueueSize);
```

Comment:
Can we change the Queue constructor to something like:

```js
new Queue(() => this.buildNewApplicationInstance(), totalSize)
```

Then the queue would "fill" itself.